### DISCUSSION TITLE
How to use LTree with SQLModel

### DISCUSSION THREAD

**Discussion by Lokeesh-27** [fastapi/sqlmodel#766] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional
from sqlmodel import Field, SQLModel

class HeroCategory(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str
    path: str  # Need to define the correct data type here
```

### Description

⚙️ **Challenge**: The 'path' column in this model is intended to store hierarchical data, and I plan to use PostgreSQL's ltree data type for efficient querying. However, I'm uncertain about the correct data type to use in the SQLModel for the 'path' column that aligns with PostgreSQL's ltree.

📚 **Reference**: [PostgreSQL ltree Documentation](https://www.postgresql.org/docs/current/ltree.html)

🎯 **Question**:

1. What is the appropriate data type for the 'path' column in the SQLModel to ensure compatibility with PostgreSQL's ltree?
2. How can I effectively query this 'path' column in SQLModel, particularly for filtering based on hierarchical data?

Looking forward to insights and suggestions from the community!

### Operating System

macOS

### SQLModel Version

0.0.8

### Python Version

3.10

**Comment by cerealkill** [comment]
I have been very divided by using `sqlmodel`. It is a neat tool, but some basic stuff is missing - it does not support Array fields like `text[]` for example, and it does not give the user a simple way to extend the lib to add what it needed. `ltree` is a great example of a downfall. I am trying to have a where clause with a ltree expression and sqlmodel does not provide a simple way to do that. example: `.where(Item.path @> path_str)`

**Reply by cerealkill** [reply]
UPDATE: I was able to send direct `where` statements by using the following code:
```python
from sqlalchemy import text
select(Item)
.where(text(f"item.path @> '{path_str}'")
```
So please document this in `sqlmodel` documentation.

**PS: Array is still an issue I could not find a workaround.**
